client/src/pages/AddProject.js

```diff
@@ -0,0 +1,227 @@
+                validWhen: false,
+                message: 'Title is required.'
+            },
+            {
+                field: 'industry',
+                method: validator.isEmpty,
+                validWhen: false,
+                message: 'Industry is required'
+            },
+            {
+                field: 'location',
+                method: validator.isEmpty,
+                validWhen: false,
+                message: 'location is required'
+            },
+            {
+                field: 'fundingGoal',
+                method: validator.isNumeric,
+                validWhen: true,
+            }
+        ]);
+        this.state = {
+            uploading: false,
+            title: '',
+            subtitle: '',
+            industry: '',
+            location: '',
+            images: [],
+            fundingGoal: 0,
+            formData: {},
+            validation: this.validators.valid()
+        }
+    }
+
+    componentDidMount() {
+        this.setState({ formData: new FormData() })
+    }
+
+    handleInput = (event) => {
+        const { value, name } = event.target;
+        const { formData } = this.state;
+        formData.set(name, value);
+        this.setState({ [name]: value, formData })
+    }
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        const validation = this.validators.validate(this.state);
+        this.setState({ validation });
+
+        if (validation.isValid) {
+            const { formData } = this.state;
+            const { id } = this.props.match.params; // will have to change eventually.
+            const newProject = await addProject(id, formData);
+            console.log(newProject);
+            this.props.history.push('/profile');
+            this.props.activateToast('Upload Successful', 'success');
+        }
+    }
+
+    setImages = (newImages) => {
+        const { formData } = this.state;
+        const addImage = [...this.state.images, ...newImages];
+        for (const image of addImage) {
+            formData.append('images', image);
+        }
+        formData.append('images', addImage);
+        this.setState({ images: addImage });
+    }
+
+    disableSubmit = () => {
+        const { title, industry, location } = this.state;
+
+        return [title, industry, location].some((value) => !value);
+    };
+
+    render() {
+        const { classes } = this.props;
+        const { title, subtitle, industry, location, images, fundingGoal, validation } = this.state;
+        return (
+            <main className={classes.pageContent}>
+                {/* <div className={classes.projectPreview}>
+                    <h2>Product Preview Thing</h2>
+                    <Button type="submit" variant="contained" color="primary" >
+                        Preview
+                    </Button>
+                </div> */}
+                <div className={classes.addProjectPage}>
+                    <h2>Start with basics</h2>
```

Comment:
Also I created a component for the header, as I think I recall it looking similar to login/register pages: CenteredPageHeader

Reply:
Ah I opted not to go with it because I just wanted the text without the green border